net: use cached peername to resolve remote props

jameshartig opened the pull request:
Allows socket.remote\* properties to still be accessed even after the
socket is closed. Fixes #9287

cjihrig commented on test/simple/test-net-remote-address-port.js:
```diff
@@ -53,12 +57,20 @@ server.listen(common.PORT, 'localhost', function() {
     client.end();
   });
+  client.on('close', function() {
+    assert.notEqual(-1, remoteAddrCandidates.indexOf(client2.remoteAddress));
```
Should these two lines be checking `client` instead of `client2`?

fastest963 replied:
Good catch. Woops!

cjihrig commented:
One question about the test. I haven't run the code, but if the CI is happy, then this LGTM.

Commit:
net: use cached peername to resolve remote props

Allows socket.remote* properties to still be accessed even after the
socket is closed. Fixes #9287

fastest963 commented:
Good catch. Fixed the test and it still passes.

cjihrig commented:
Thanks! Landed in8c38b07252d86f4eef91344f8df4f0c3e38bca35.

jasnell commented:
@cjihrig ... do you happen to know why this was reopened?

cjihrig commented:
I think that was a malfunction in the TJ Fontaine bot.
